e2e/specs/st_multiselect.spec.js

```diff
@@ -240,9 +250,41 @@ describe("st.multiselect", () => {
             "value 6: []" +
             "value 7: []" +
             "value 8: []" +
-            "value 9: ['male']" +
+            "value 9: []" +
+            "value 10: ['male']" +
             "multiselect changed: True"
         );
     });
   });
+
+  describe("when using max_selections for st.multiselect", () => {
+    it("should show the correct text when maxSelections is reached", () => {
+      selectNthOptionForKthMultiselect(0, 8)
+      cy.getIndexed(".stMultiSelect", 8).then(el => {
+        cy
+          .wrap(el)
+          // recheck that we have hit the limit
+          .find("input")
+          .click()
+          .get("li")
+          .first()
+          .should("have.text", "You can only select up to 1 options. Remove an option first.")
+      });
+    });
+
+
+    it("should display an error when options > max selections set during session state", () => {
+      cy.get(".stCheckbox")
+        .first()
+        // For whatever reason both click() and click({ force: true }) don't want
+        // to work here, so we use {multiple: true} even though we only take a
+        // snapshot of one of the checkboxes below.
+        .click({ multiple: true });
+
+      cy.get(".element-container .stException").should(
+        "contain.text",
+        `Multiselect has 2 options selected but max_selections\nis set to 1. This happened because you either gave too many options to default\nor you manipulated the widget's state through st.session_state. Note that\nthe latter can happen before the line indicated in the traceback.\nPlease select at most 1 options.`
```

Comment:
nit: We can probably just match against the first line of the error message here to avoid having to duplicate the entirety of it here (we test for the full message in python unit tests anyway)

Reply:
Same note about special-casing (in `multiselect.py`) so the grammar is right when there's only 1 option for this error message